cinema/urls.py

```diff
@@ -1,10 +1,25 @@
-from django.urls import path
+from django.urls import path, include
+from rest_framework.routers import DefaultRouter
+from .views import (
+    GenreAPIView,
+    ActorListCreateAPIView,
+    ActorRetrieveUpdateDestroyAPIView,
+    CinemaHallViewSet,
+    MovieViewSet
+)
+
+app_name = 'cinema'
+
+router = DefaultRouter()
+router.register(r'cinemahalls', CinemaHallViewSet, basename='cinemahall')
+router.register(r'movies', MovieViewSet, basename='movie')
 
-from cinema.views import movie_list, movie_detail
 
 urlpatterns = [
-    path("movies/", movie_list, name="movie-list"),
-    path("movies/<int:pk>/", movie_detail, name="movie-detail"),
+    path('genres/', GenreAPIView.as_view(), name='genre-list'),
+    path('actors/', ActorListCreateAPIView.as_view(), name='actor-list-create'),
+    path('actors/<int:pk>/', ActorRetrieveUpdateDestroyAPIView.as_view(), name='actor-detail'),
+    path('', include(router.urls)),
 ]
 
-app_name = "cinema"
+urlpatterns += router.urls
```

Comment:
Including `router.urls` twice in `urlpatterns` is redundant. The line `urlpatterns += router.urls` is unnecessary since `path('', include(router.urls))` already includes all the routes from the router.